### PR TITLE
chore: use build artifacts

### PR DESCRIPTION
**Pull request by LayZeeDK**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
1. Cache is being used for the packaged library.
1. The `app` and `e2e` jobs have to wait for the `lib` job.
1. Angular 8.1 is listed as supported.

Issue Number: N/A

## What is the new behavior?
1. The packaged library is uploaded as a build artifact in the newly created `build` job.
1. The `app` and `e2e` can start right after the `build` job as they download the build artifact with the packaged library.
1. Angular 8.1 is removed from the CI pipeline as it doesn't support our bundle.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information
Deprecates #33.

### CODE REVIEW

**Comment by NachoVazquez** [comment]
The difference between cache and upload artifacts is basically the availability of the data over time. Not sure we need anything more than cache.

https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows#comparing-artifacts-and-dependency-caching

**Comment by LayZeeDK** [comment]
Caching is a lot more complex in the pipeline though. Artifacts give us the ability to download a build and try it out locally. But I'm struggling to get Angular v8 working for some reason which makes me question or current CI with v8 (we forgot v8 in `e2e`, besides we're not even verifying the library in our `app` and `e2e` tests), so I'll do some manual local tests to try to figure all of this out so that we can have confidence and simpler pipelines.

Even if we want to continue using the cache rather than build artifacts, we should extract a build step to allow the full pipeline to finish faster.

**Comment by LayZeeDK** [comment]
The cache action does a lot of additional work after each job that uses it. This makes build artifacts faster as they're only stored once.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -61,40 +61,67 @@ jobs:
         with:
           angular_version: ${{ matrix.angular-version }}
 
-      - name: Get yarn cache directory path
+      - name: Variabæe-Yarn cache directory path
```
Comment (NachoVazquez) [inline .github/workflows/ci.yml:64]:
```suggestion
      - name: Variable-Yarn cache directory path
```

**Pull request merged**